File: back/InterViewBacked/src/main/java/com/wjh/interviewbacked/entity/Note.java
package com.wjh.interviewbacked.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 笔记元数据实体，对应数据库 note 表（不含正文，正文在 note_content）。
 * tags 为 JSON 字符串数组（展示冗余；管理真相源见 note_tag_rel）。
 */
@Data
public class Note {
    private String id;
    /** 数据归属（消除水平越权） */
    private String userId;
    private String title;
    private String summary;
    /** 分类 id：frontend / backend / testing / tools / interview */
    private String category;
    /** 标签 JSON 数组（冗余展示） */
    private String tags;
    private Boolean pinned;
    private Integer views;
    /** 乐观锁版本号，防并发覆盖 */
    private Integer version;
    /** 正文 MD5，后端去重（拦截重复写库） */
    private String contentHash;
    /** 可见性：0 私有（仅作者本人可见） 1 讨论广场公开（所有可见） 2 链接分享（预留） */
    private Integer visibility;
    /** 软删除：0 正常 1 回收站 */
    private Integer deleted;
    /** 软删除时间，用于 30 天自动清理 */
    private LocalDateTime deletedAt;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    /* ========== 以下为非数据库字段，仅用于联表查询回填 ========== */
    /** 作者 id（广场列表冗余，便于前端跳主页） */
    private String authorId;
    /** 作者昵称（广场列表 JOIN user.nickname 回填） */
    private String authorName;
    /** 全文检索相关度得分（仅知识库问答召回时回填） */
    private Double score;
}

File: front/InterviewProjectProject/src/types/note.ts
/**
 * 笔记领域模型
 * 字段与后端 NoteDTO 保持一致
 */

/** 笔记可见性：0 私有（仅作者本人） 1 讨论广场公开 */
export type NoteVisibility = 0 | 1

/** 列表数据来源：mine = 我的笔记，plaza = 讨论广场 */
export type NoteScope = 'mine' | 'plaza'

export interface Note {
  id: string
  title: string
  /** 列表页展示的摘要，不填时由正文自动截取 */
  summary: string
  /** Markdown 正文（列表接口不返回，需单独拉详情） */
  content: string
  /** 所属分类 id */
  category: string
  tags: string[]
  pinned: boolean
  views: number
  /** 可见性：0 私有 1 广场公开 */
  visibility: NoteVisibility
  /** 作者 id（后端一定会回填，便于前端判断「是不是我的笔记」） */
  authorId?: string
  /** 作者昵称（广场列表返回） */
  authorName?: string
  createdAt: string
  updatedAt: string
}

/** 新建 / 更新笔记的入参（不含服务端生成的字段） */
export type NoteDraft = Omit<
  Note,
  'id' | 'views' | 'createdAt' | 'updatedAt' | 'visibility' | 'authorId' | 'authorName'
> & {
  /** 是否发布到讨论广场：true 公开给所有人；false 仅自己可见。不传表示不改动可见性 */
  toPlaza?: boolean
}

export interface NoteCategory {
  id: string
  label: string
  icon: string
}

export interface NoteQuery {
  keyword?: string
  category?: string
  tag?: string
}
